### PR TITLE
Streamline error codes

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area TODO
/kind TODO

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: pkg/apis/core/v1beta1/helper/errors.go [hunk 1]
```diff
@@ -52,7 +52,8 @@ func (e *ErrorWithCodes) Error() string {
 }
 
 var (
-	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|InvalidAuthenticationTokenTenant|SignatureDoesNotMatch|Authentication failed|AuthFailure|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId|AccessDenied|OperationNotAllowed|Error 403)`)
+	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|SignatureDoesNotMatch|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId|AccessDenied|OperationNotAllowed|Error 403)`)
+	unauthenticatedRegexp               = regexp.MustCompile(`(?i)(InvalidAuthenticationTokenTenant|Authentication failed|AuthFailure)`)
```
Comment (shafeeqes) [inline pkg/apis/core/v1beta1/helper/errors.go]: did I pick the right errors? should I move any others from unauthorized?

File: pkg/apis/core/types_common.go [hunk 2]
```diff
@@ -25,9 +25,9 @@ const (
 	// ErrorInfraUnauthorized indicates that the last error occurred due to invalid infrastructure credentials.
 	// It is classified as a non-retryable error code.
 	ErrorInfraUnauthorized ErrorCode = "ERR_INFRA_UNAUTHORIZED"
-	// ErrorInfraInsufficientPrivileges indicates that the last error occurred due to insufficient infrastructure privileges.
+	// ErrorInfraUnauthenticated indicates that the last error occurred due to invalid authentication details.
```
Comment (shafeeqes) [inline pkg/apis/core/types_common.go]: any changes needed to the doc string?

File: pkg/apis/core/types_common.go [hunk 3]
```diff
@@ -25,9 +25,9 @@ const (
 	// ErrorInfraUnauthorized indicates that the last error occurred due to invalid infrastructure credentials.
```
Comment (shafeeqes) [inline pkg/apis/core/types_common.go]: any changes needed to the doc string?

**Comment by BeckerMax** [comment]
Please add a description and reference the issue you want to solve in the PR description. Generally PRs and issues serve as a help for understanding code changes for future readers.

**Comment by BeckerMax** [comment]
In commit messages please try to always follow the imperative style: https://cbea.ms/git-commit/#imperative

**Comment by BeckerMax** [comment]
And you need to open a PR to the master of gardener/gardener currently you opened a PR to the master of your fork

**Comment by shafeeqes** [comment]
Yes yes. I just did this as a draft, in the actual PR, I will mention all that.

File: pkg/apis/core/v1beta1/helper/errors.go (hunk 1 above)
Comment (shafeeqes) [reply pkg/apis/core/v1beta1/helper/errors.go]: I guess we should probably add InvalidAccessKeyId|InvalidSecretAccessKey|InvalidClientTokenId|InvalidSubscriptionId to Unauthenticated error code. WDYT?

Comment (BeckerMax) [reply pkg/apis/core/v1beta1/helper/errors.go]: Most of them look fine. Please google the error strings you are unsure of and check for example here:
https://docs.aws.amazon.com/STS/latest/APIReference/CommonErrors.html
See also https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses

File: pkg/apis/core/types_common.go (hunk 2 above)
Comment (BeckerMax) [reply pkg/apis/core/types_common.go]: actually I personally like a definition close to the HTTP error code. 
Unauthenticated -> https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401
Unauthorized -> https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403
That is my personal preference. Generally the doc strings look ok.

**Comment by BeckerMax** [comment]
Apart from the comments above PR looks good to be opened.

**Pull request closed without merging**